### ISSUE TITLE
provide current slide element on event  argument

### ISSUE THREAD

**Issue by djangosdk** [sachinchoolur/lightGallery#87]
Hi,
I was reading events from docs and I feel need for the following.
event must provide clicked element  and slide element.
For example, 
```
    <ul id="lightGallery">
    <li data-html="#video1">
       <img src="img/thumb1.jpg" />
    </li>
    </ul>

onSlideBefore : function(item,slide) {
// where item should be DOM li with relevant data-src
// slide Dom element which is in focus at the moment
}, 
```
Also addition off ```onSlide```event would be nice , this way users will not have to create html element with id=  #video1 and they can return their custom html element to slide DOm element 
for example 
```
    <ul id="lightGallery">
    <li >
       <img src="img/thumb1.jpg" />
    </li>
    </ul>

onSlide : function(item,slide) {
//create html
 var vid = '<video> </video>'';
// append to slide
 slide.append(vid ); 
}
```

**Comment by sachinchoolur** [comment]
I think you can use  `onSlideBefore() ` function which will execute immediately before each slide transition.
I will add `slide` parameter for `onSlideBefore(slide)` callback function in the next version.

**Comment by djangosdk** [comment]
thank you.
Also if some way is devised that one doest not have to create so much slides in html would be nice.
JS should create slide HTMl for users.
would be nice to have

**Comment by sachinchoolur** [comment]
Hi,
I have released lightgallery version 1.2.
[Here](http://sachinchoolur.github.io/lightGallery/docs/api.html#events) you can see more information about events..
